python/version_details/version_details.py

```diff
@@ -520,6 +520,10 @@ def load_data(self, entity):
             self.clear()
             return
 
+        # Ensure project schema will be loaded. The project schema is needed
+        # to reference Version fields in various UI and other elements.
+        shotgun_globals.run_on_schema_loaded(lambda : None, entity.get("project"))
```

Comment:
If you look at the run_on_schema_loaded implementation

```
  def run_on_schema_loaded(cls, callback, project_id=None):
        Run the given callback once the schema is loaded.

        :param callback:    Method with no argument to run when the schema is loaded
        :param project_id:  The id of the project entity to load the schema for. If
                            None, the current context's project will be used.

        self = cls.__get_instance()

        if self._is_schema_loaded(project_id=project_id):
            callback()
        else:
            self.schema_loaded.connect(callback)

            # kick off full schema loading
            self._check_schema_refresh(project_id=project_id)

```
You see that the code will kick off the schema loading in the else case. So this code isn't expecting to block or wait for the load to happen. It will just trigger the cache to be loaded sooner. Which will help make the data available to the UI sooner, before it is populated. The lambda does nothing probably because @eliveikis was interested in the triggering the cache to be populated.

What's not so clear to me is there is code that will synchronize and guarantee that the UI will be populated always AFTER the cache has populated. Any insights into how that happens @eliveikis?